File: product-web-app/service/productValidator.ts
import Product from '../model/product.js';

export default class ProductValidator {
  validateProduct(product: Product) {
    if (product.name.length > 50) {
      return 'Name greater than 50 characters';
    }

    if (product.description.length > 500) {
      return 'Description greater than 500 characters';
    }

    if (product.price < 10) {
      return 'Price less than 10 ZŁ';
    }

    return null;
  }

  validateSomethingElse() {
    const test = 'abc';

    return test + test;
  }
}

File: product-backend/src/main/java/org/kainos/ea/service/ProductValidator.java
package org.kainos.ea.service;

import org.kainos.ea.model.ProductRequest;

public class ProductValidator {
    public String isValidProduct(ProductRequest product) {

        if (product.getName().length() > 75) {
            return "Name greater than 75 characters";
        }

        if (product.getDescription().length() > 500) {
            return "Description greater than 500 characters";
        }

        if (product.getPrice() < 10) {
            return "Price less than £10";
        }

        return null;
    }
}
